graphql-rewrite/src/common.rs

```diff
@@ -0,0 +1,335 @@
+use std::{fmt, collections::BTreeMap};
+
+use combine::{parser, ParseResult, Parser};
+use combine::easy::Error;
+use combine::error::StreamError;
+use combine::combinator::{many, many1, optional, position, choice};
+use num_bigint::BigInt;
+use num_traits::cast::ToPrimitive;
+
+use crate::tokenizer::{Kind as T, Token, TokenStream};
+use crate::helpers::{punct, ident, kind, name};
+use crate::position::Pos;
+
+/// Text abstracts over types that hold a string value.
+/// It is used to make the AST generic over the string type.
+pub trait Text<'a>: 'a {
+    type Value: 'a + From<&'a str> + AsRef<str> + std::borrow::Borrow<str> + PartialEq + Eq + PartialOrd + Ord + fmt::Debug + Clone;
```

Comment:
You typically limit line length to 80, is there a reason why this line is longer? (just trying to understand the style)

Reply:
Well, this is contributed code (to graphql-parser).